src/QsCompiler/CompilationManager/CompilationUnit.cs

```diff
@@ -51,9 +50,9 @@ public class Headers
             public Headers(string source, IEnumerable<QsNamespace> syntaxTree)
                 : this(
                     source,
-                    syntaxTree.Callables().Where(c => c.SourceFile.EndsWith(".qs")).Select(CallableDeclarationHeader.New),
-                    syntaxTree.Specializations().Where(c => c.SourceFile.EndsWith(".qs")).Select(s => (SpecializationDeclarationHeader.New(s), (SpecializationImplementation?)s.Implementation)),
-                    syntaxTree.Types().Where(c => c.SourceFile.EndsWith(".qs")).Select(TypeDeclarationHeader.New))
+                    syntaxTree.Callables().Where(c => c.Source.AssemblyFile.IsNull).Select(CallableDeclarationHeader.New),
```

Comment:
Does this mean that any `Source` with a null `AssemblyFile` is a Q# source?

Filtering based on this condition makes this code a bit harder to understand. Would it be possible to add a property to `Source` whose name is more explicit (e.g. `IsQSharp`)?

Reply:
`IsQSharp` seems weird to me - all sources are Q# sources. Maybe naming the opposite is easier, `IsReference`?

Reply:
`IsReference` sounds good to me.